### PR TITLE
New Event: PlayerPickupArrowEvent

### PR DESCRIPTION
**Pull request by DelayedGaming**
### Description
This PR adds [PlayerPickupArrowEvent](https://jd.papermc.io/paper/1.19/org/bukkit/event/player/PlayerPickupArrowEvent.html) to Skript!

### Note
`event-projectile` will not return anything for 1.13 users due to an API change in 1.14 onwards with a different return type at the event constructor. This decision was [discussed and approved](https://discord.com/channels/135877399391764480/836220422223036467/1094263795582521354) by the SkriptLang team.

---
**Target Minecraft Versions:** any
**Requirements:** Minecraft 1.14 (event-projectile)
**Related Issues:** https://github.com/SkriptLang/Skript/issues/4788

### CODE REVIEW

File: src/main/java/ch/njol/skript/classes/data/BukkitEventValues.java [hunk 1]
```diff
@@ -1562,5 +1564,23 @@ public Location get(LootGenerateEvent event) {
 				}
 			}, EventValues.TIME_NOW);
 		}
+
+		// PlayerPickupArrowEvent
+		if (Skript.isRunningMinecraft(1, 14)) {
```
Comment (AyhamAl-Ali) [inline src/main/java/ch/njol/skript/classes/data/BukkitEventValues.java:1569]: Add a note for future developers about Arrow and AbstractArrow issue :)

File: src/main/java/ch/njol/skript/classes/data/BukkitEventValues.java [hunk 2]
```diff
@@ -1562,5 +1564,23 @@ public Location get(LootGenerateEvent event) {
...
+			EventValues.registerEventValue(PlayerPickupArrowEvent.class, AbstractArrow.class, new Getter<AbstractArrow, PlayerPickupArrowEvent>() {
+				@Override
+				public AbstractArrow get(PlayerPickupArrowEvent event) {
```
Comment (AyhamAl-Ali) [inline src/main/java/ch/njol/skript/classes/data/BukkitEventValues.java:1572]: Is it okay to make this return Projectile?

**Review by AyhamAl-Ali — APPROVED** [review]
Noice ⚡

File: src/main/java/ch/njol/skript/events/SimpleEvents.java [hunk 3]
```diff
@@ -709,6 +710,15 @@ public class SimpleEvents {
 					.requiredPlugins("Paper 1.16+");
 		}
 
+			Skript.registerEvent("Player Pickup Arrow", SimpleEvent.class, PlayerPickupArrowEvent.class, "[player] (pick[ing| ]up [an] arrow|arrow pick[ing| ]up)")
```
Comment (TheLimeGlass) [inline src/main/java/ch/njol/skript/events/SimpleEvents.java:713]:
```suggestion
			Skript.registerEvent("Player Pickup Arrow", SimpleEvent.class, PlayerPickupArrowEvent.class, "[player] (pick[ing| ]up [of an] arrow|arrow pick[ing| ]up)")
```

**Review by TheLimeGlass — APPROVED** [review]

File: src/main/java/ch/njol/skript/events/SimpleEvents.java (hunk 3 above)
Comment (DelayedGaming) [reply src/main/java/ch/njol/skript/events/SimpleEvents.java:713]: Can we make the keyword `of` optional?

Comment (AyhamAl-Ali) [reply src/main/java/ch/njol/skript/events/SimpleEvents.java:713]: IMO it should be optional as with and without is correct

Comment (TheLimeGlass) [reply src/main/java/ch/njol/skript/events/SimpleEvents.java:713]: On pickup an arrow is not proper

Comment (DelayedGaming) [reply src/main/java/ch/njol/skript/events/SimpleEvents.java:713]: Thats fair, but `on picking up an arrow` is correct too

Comment (AyhamAl-Ali) [reply src/main/java/ch/njol/skript/events/SimpleEvents.java:713]: > On pickup an arrow is not proper

That's fine, we can't create syntaxes for all patterns

**Pull request merged**